onnxruntime/core/providers/cuda/object_detection/roialign.cc

```diff
@@ -57,7 +57,7 @@ Status RoiAlign<T>::ComputeInternal(OpKernelContext* context) const {
         this->sampling_ratio_,
         reinterpret_cast<const typename ToCudaType<T>::MappedType*>(rois_ptr->Data<T>()),
         num_roi_cols,
-        reinterpret_cast<typename ToCudaType<T>::MappedType*>(Y.MutableData<T>()),
+        reinterpret_cast<typename ToCudaType<T>::MappedType*>(Y.template MutableData<T>()),
         this->mode_ == RoiAlignMode::avg,
```

Comment:
Please do not add unnecessary 'template ' qualifiers as they add unnecessary noise to the code. They are only needed if the 'Tensor' type is one of the templatized types (in this case if T == Tensor and Y was of type T you would need it).